server/client.go

```diff
@@ -1226,6 +1227,15 @@ func (c *client) readLoop(pre []byte) {
 		// to process messages, etc.
 		for i := 0; i < len(bufs); i++ {
 			if err := c.parse(bufs[i]); err != nil {
+				if err == ErrMinimumVersionRequired {
+					// Special case here, currently only for leaf node connections.
+					// When process the CONNECT protocol, if the minimum version
+					// required was not met, an error was printed and sent back to
+					// the remote, and connection was closed after a certain delay
+					// (to avoid "rapid" reconnection from the remote).
+					// We don't need to do any of the things below, simply return.
+					return
```

Comment:
should there be a trace here? so it's printed on either end.

Reply:
In processLeafNodeConnect we have:
```
c.sendErrAndErr(fmt.Sprintf("connection rejected since minimum version required is %q", mv))
```
which means that the accept side logs AND sends that error to the remote. The remote will see that:
```
[55435] 2022/04/07 12:11:29.131489 [INF] 127.0.0.1:7222 - lid:4 - Leafnode connection created for account: $G 
[55435] 2022/04/07 12:11:29.132505 [INF] 127.0.0.1:7222 - lid:4 - JetStream Not Extended, adding deny [$JS.API.> $KV.> $OBJ.>] for account "$G"
[55435] 2022/04/07 12:11:29.132580 [ERR] 127.0.0.1:7222 - lid:4 - Leafnode Error 'connection rejected since minimum version required is "2.9.0"'
[55435] 2022/04/07 12:11:34.133252 [INF] 127.0.0.1:7222 - lid:4 - Leafnode connection closed: Read Error account: $G
```
Again, see the read error (due to other side closing the connection) 5 seconds after being notified of the error. This is intentional so that the remote does not reconnect "right away"

Reply:
ok, so already logged before getting to this return

Reply:
Yes.